modules/nf-core/sourmash/taxannotate/main.nf

```diff
@@ -0,0 +1,43 @@
+process SOURMASH_TAXANNOTATE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::sourmash=4.6.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.6.1--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.6.1--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(gather_results)
+    path(taxonomy)
+
+    output:
+    tuple val(meta), path("*.with-lineages.csv.gz"), emit: result
+    path "versions.yml"                            , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args   = task.ext.args ?: ''
+    prefix="${gather_results.getBaseName().replaceAll(/.csv/, '')}"
+
+    """
+    sourmash \\
+        tax annotate \
+        $args \\
+        --gather-csv ${gather_results} \\
+        --taxonomy ${taxonomy} \\
+        --output-dir "."
+
+    ## Output file name = "${prefix}".with-lineages.csv
```

Comment:
```suggestion
    ## Output file name = "${gather_results}".with-lineages.csv
```

right? as this is what the tool actually generates.

However in this case I would rather not call it `prefix` as that should be reserved a user can customise via `ext.args` (which is not being used here). 

Actually do we need any of this inferring the file name at all? Are there any other ending in `with-lineages.csv`? Otherwise would simply not just after the annotate command, you run `gzip --no-name *.with-lineages.csv` ? No need to specify the full name?

Reply:
Given `gather_results` = `tmp.csv.gz`, sourmash will produce `tmp.with-lineages.csv`, 
so `${gather_results}.with-lineages.csv` is different - it will be expanded as `tmp.csv.gz.with-lineages.csv`.
So if we want to name the output file explicitly, we must use `${prefix}.with-lineages.csv`.

However, I agree that there should be just a single `*.with-lineages.csv` file, so no need in the `prefix` at all.
I will fix the code in a minute.

Reply:
I see, OK! I'm going to give you the approval - sorry for complicating things!